test/unit/cute/intel_xe/copy_block.cpp

```diff
@@ -0,0 +1,1817 @@
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ **************************************************************************************************/
+
+#include <cute/tensor.hpp>
+#include <sycl/sycl.hpp>
+#include <syclcompat.hpp>
+
+#include "cutlass_unit_test.h"
+
+using namespace cute;
+using namespace cutlass;
+
+template <class TensorS, class TensorD, class TiledLoad, class TiledStore, class CopyOp = void>
+void copy_kernel_vectorized(TensorS S, TensorD D, TiledLoad load,
+                            TiledStore store) {
+  const int m_coord = 0;
+  const int n_coord = 0;
+  const int l_coord = BlockIdxZ();
+
+  // ==========  load   ==========
+  auto thr_copy_load = load.get_thread_slice(syclcompat::local_id::x());
+  auto thr_tile_load_D = thr_copy_load.partition_D(S);
+  auto fragment = make_fragment_like(thr_tile_load_D);
+  auto ld_tensor =
+      load.get_pvc_tensor(make_coord(m_coord, n_coord, l_coord),
+                          fragment.shape(), typename TiledLoad::Shape_MN{});
+  if constexpr (cute::detail::has_prefetch<CopyOp>) prefetch(load, ld_tensor);
+  copy(load, ld_tensor, fragment);
+
+  // ==========  store   ==========
+  auto thr_copy_store = store.get_thread_slice(syclcompat::local_id::x());
+  Tensor frag_view =
+      make_tensor(static_cast<decltype(fragment) &&>(fragment).data(),
+                  thr_copy_store.partition_S(D).shape());
+  auto st_tensor = store.get_pvc_tensor(make_coord(m_coord, n_coord, l_coord),
+                                        frag_view.shape(),
+                                        typename TiledStore::Shape_MN{});
+  copy(store, frag_view, st_tensor);
+
+#if 1
+  if (thread(1)) {
+    print("fragment: ");
+    print(fragment.layout());
+    print("\n");
+
+    print("ld_tensor: ");
+    print(ld_tensor.layout());
+    print("\n");
+
+    print("frag_view: ");
+    print(frag_view.layout());
+    print("\n");
+
+    print("st_tensor: ");
+    print(st_tensor.layout());
+    print("\n");
+  }
+#endif
+}
+
+TEST(PVC_2d_copy, load_store) {
```

Comment:
Could you create separate functions for each test case instead of having one single function that covers all of them?